nemo_curator/utils/file_utils.py

```diff
@@ -45,7 +46,44 @@ def expand_outdir_and_mkdir(outdir):
     return outdir
 
 
-def get_all_files_paths_under(root, recurse_subdirectories=True, followlinks=False):
+def filter_files_by_extension(
+    files_list: List[str],
+    filter_by: Union[str, List[str]],
```

Comment:
nit : I think have a more verbose arg name might be beneficial here such as `keep_extensions` or `extensions_list`, since `filter` is rather open-ended, and when I first read through I thought it is a list of filters on filename (e.g. UDF like `lambda file_name : True if .... else False`)

Reply:
nit2 : having return type hint

Reply:
Good point, thanks!